edit-post/components/sidebar/post-taxonomies/taxonomy-panel.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External Dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/element';
+import { PanelBody } from '@wordpress/components';
+import { withSelect, withDispatch } from '@wordpress/data';
+
+function TaxonomyPanel( { taxonomy, isOpened, onTogglePanel, children } ) {
+	const taxonomyMenuName = get( taxonomy, [ 'labels', 'menu_name' ] );
+	if ( ! taxonomyMenuName ) {
+		return null;
+	}
+	return (
+		<PanelBody
+			title={ taxonomyMenuName }
+			opened={ isOpened }
+			onToggle={ onTogglePanel }
+		>
+			{ children }
+		</PanelBody>
+	);
+}
+
+export default compose(
+	withSelect( ( select, ownProps ) => {
+		const slug = get( ownProps.taxonomy, [ 'slug' ] );
+		const panelName = slug ? `taxonomy-panel-${ slug }` : '';
+		return {
+			panelName,
+			isOpened: slug ?
+				select( 'core/edit-post' ).isEditorSidebarPanelOpened( panelName ) :
+				false,
+		};
+	} ),
+	withDispatch( ( dispatch, ownProps ) => ( {
+		onTogglePanel: () => {
+			if ( ownProps.panelName ) {
+				return dispatch( 'core/edit-post' ).
+					toggleGeneralSidebarEditorPanel( ownProps.panelName );
```

Comment:
Minor: If we're concerned about line length, we could invert the condition to an early return to avoid the tab.

```js
if ( ! ownProps.panelName ) {
	return;
}

dispatch( 'core/edit-post' ).toggleGeneralSidebarEditorPanel( ownProps.panelName );
```